### PR TITLE
Loans pallet cleanup

### PR DESCRIPTION
**Pull request by offerijns**
Resolves https://github.com/centrifuge/centrifuge-chain/issues/720 and part of https://github.com/centrifuge/centrifuge-chain/issues/717

### CODE REVIEW

File: pallets/loans/src/functions.rs [hunk 1]
```diff
@@ -70,20 +71,16 @@ impl<T: Config> Pallet<T> {
 		);
 
 		// create new loan nft
-		let loan_pallet_account: T::AccountId = T::LoansPalletId::get().into_account();
```
Comment (mustermeiszer) [inline pallets/loans/src/functions.rs:73]: Is the `LoansPalletId` used anywhere else to hold something or can we remove it?

File: pallets/loans/src/functions.rs [hunk 2]
```diff
@@ -224,27 +213,24 @@ impl<T: Config> Pallet<T> {
...
 		Self::check_loan_owner(pool_id, loan_id, owner.clone())?;
 
-		LoanInfo::<T>::try_mutate(
+		Loan::<T>::try_mutate(
 			pool_id,
 			loan_id,
-			|maybe_loan_info| -> Result<bool, DispatchError> {
-				let mut loan_info = maybe_loan_info.take().ok_or(Error::<T>::MissingLoan)?;
+			|maybe_loan| -> Result<bool, DispatchError> {
+				let mut loan = maybe_loan.take().ok_or(Error::<T>::MissingLoan)?;
```
Comment (mustermeiszer) [inline pallets/loans/src/functions.rs:220]: I would love to get rid of all the `let mut loan = maybe_loan.take().ok_or(Error::<T>::MissingLoan)?;` and rather work on the references instead via `let mut loan = maybe_loan.as_ref().ok_or(Error::<T>::MissingLoan)?;`.

Replacing the given Option with a None feels bad.

**Review by mustermeiszer — APPROVED** [review]
LGTM! 

Just one nity bit, that I don`t like the logic that we currently have here in most of the `try_mutate` calls, where we are taking the possible `Some(T)` and then storing it at the end of the logic again in the `Option<t>`

File: pallets/loans/src/functions.rs (hunk 2 above)
Comment (branan) [reply pallets/loans/src/functions.rs:220]: I agree, I much prefer the `as_mut()` pattern we use in the `Pools` pallet

File: pallets/loans/src/functions.rs (hunk 1 above)
Comment (offerijns) [reply pallets/loans/src/functions.rs:73]: It's used here: https://github.com/centrifuge/centrifuge-chain/blob/loans-cleanup/pallets/loans/src/functions.rs#L38 Currently the loan NFTs are transferred here, and it's also used by the tests to find the pallet account. We can probably consider removing this, but I'd rather do that in a separate PR.

File: pallets/loans/src/functions.rs (hunk 2 above)
Comment (offerijns) [reply pallets/loans/src/functions.rs]: Just changed this @mustermeiszer @branan

**Review by mustermeiszer — APPROVED** [review]
Nice! LGTM!

**Comment by branan** [comment]
I'm happy with this now, as well. Will hit the button when CI completes

**Pull request merged**